src/core/mesh/qgsmeshlayerutils.h

```diff
@@ -282,6 +282,14 @@ class CORE_EXPORT QgsMeshLayerUtils
       const QgsTriangularMesh &triangularMesh,
       const QVector<double> &verticalMagnitude,
       bool isRelative );
+
+    /**
+     * Changes scalar settings for classified scalar value
+     * \param meta the metadata where classification information is
+     * \param scalarSettings scalar settings that have to be modified
+     * \since QGIS 3.14
+     */
+    static void applyClassificationOnScalarSettings( const QgsMeshDatasetGroupMetadata &meta, QgsMeshRendererScalarSettings &scalarSettings );
```

Comment:
Do we need to have it in public api? I suggest to move it as private method in qgsmeshlayer.cpp